modules/EmergencyMedicine/src/EmergencyMedicine/Controller/BaseController.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace EmergencyMedicine\Controller;
+
+use FhirAPI\FhirRestApiBuilder\Parts\ErrorCodes;
+use GenericTools\Controller\BaseController as GenericBaseController;
+use Interop\Container\ContainerInterface;
+use GenericTools\Traits\saveDocToServer;
+
+class BaseController extends GenericBaseController
+{
+    use saveDocToServer;
+
+    const HEADER_PATH = 'emergency-medicine/letter-generator/letter-generator-header';
+    const FOOTER_PATH = 'emergency-medicine/letter-generator/letter-generator-footer';
+    const DOC_TYPE = "file_url";
+    const PDF_MINE_TYPE = "application/pdf";
+
+    private $container;
+
+    public function __construct(ContainerInterface $container)
+    {
+        parent::__construct($container);
+        $this->container = $container;
+    }
+
+    /**
+     * get facility data
+     */
+    public function getFacilityInfo($id = null)
+    {
+        if (!is_null($id)) {
+            $info = $this->container->get('GenericTools\Model\FacilityTable')->getFacility(intval($id));
+            $data = array();
+            $data['clinic'] = $info->name;
+            $data['address'] = $info->street . " " . $info->city;
+            $data['phone'] = $info->phone;
+            $data['email'] = $info->email;
+            return $data;
+
+        } else {
+            return array();
+        }
+    }
+
+    public function saveDocToStorage($data, $fileName, $date)
+    {
+        $dataToSave = array();
+        $dataToSave['storage']['data'] = $data;
+        $dataToSave['documents']['date'] = $date;
+        $dataToSave['documents']['url'] = $fileName;
+        $rez = $this->uploadToStorage($dataToSave);
+        return $rez;
+    }
+
+    public function saveDocInfoToDb($storageSave, $configData,$pdfEncoded)
```

Comment:
Why do not move this function to GenericTools\Traits\saveDocToServer

Reply:
this is not a generic function it is very specific structured data